### DISCUSSION TITLE
Unable to run from apptainer template because apptainer PPA missing

### DISCUSSION THREAD

**Discussion by thillerson** [lima-vm/lima#2644] [Q&A]
I'm trying to run an Apptainer instance, but the standard template ([here](https://github.com/lima-vm/lima/blob/master/examples/apptainer.yaml)) seems to be failing during the Apptainer installation.

If I `sudo su` and look at the logs in `/var/log/cloud-init-output.log`, I see this error:
```
+ add-apt-repository -y ppa:apptainer/ppa
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/httplib2/__init__.py", line 1343, in _conn_request
    conn.connect()
  File "/usr/lib/python3/dist-packages/httplib2/__init__.py", line 1119, in connect
    address_info = socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/socket.py", line 963, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution
```

And at the end: `Unable to find the server at api.launchpad.net`.

It seems clear that the repo isn't in some kind of registry, but I'm not clear on what to do next... any guidance?

**Comment by afbjorklund** [comment]
The template is using the "official" PPA: https://apptainer.org/news/official-ppa-20230206/

Your error seems to be some kind of DNS issue, or possibly an outage at https://launchpad.net

Alternatives would be installing a different OS template, and/or adding [apptainer](https://github.com/apptainer/apptainer/releases) yourself?

See: https://apptainer.org/docs/admin/latest/installation.html#installation-on-linux

**Comment by thillerson** [comment]
So, inside of my lima instance, I do indeed have a problem with DNS: 
```
ping: google.com: Temporary failure in name resolution
```

Given that I'm starting from a very basic template, `template://apptainer` is this a problem to be solved in my local environment? Any ideas of where to start?

**Reply by afbjorklund** [reply]
The DNS setup of the apptainer.yaml is the same as in ubuntu.yaml, so it seems your problem is unrelated to Apptainer

**Reply by thillerson** [reply]
And potentially unrelated to Lima. I'll do some more investigation here.
